server/configHelpers.js

```diff
@@ -0,0 +1,185 @@
+ * @param {Object} options object for a bcoin/hsd client
+ * @param {bool} force - whether or not to force config creation if client
+ * can't connect
+ * @returns {bcfg.Config}
+ */
+async function createClientConfig(id, options = {}, force = false) {
+  assert(typeof id === 'string', 'Must pass an id as first paramater');
+  // assert(options, 'Must pass config options to create a client');
+
+  let clientConfig = options;
+  if (!(options instanceof Config))
+    clientConfig = getConfigFromOptions({ id, ...options });
+
+  const appConfig = loadConfig('bpanel', options);
+  const clientsDir = appConfig.str('clients-dir', 'clients');
+
+  // get full path to client configs relative to the project
+  // prefix which defaults to `~/.bpanel`
+  const clientsPath = resolve(appConfig.prefix, clientsDir);
+
+  try {
+    await testConfigOptions(clientConfig);
+  } catch (e) {
+    if (force) {
+      logger.warn(e.message);
+      logger.warn('Creating config file anyway...');
+    } else throw e;
+  }
+
+  let configTxt = '';
+  for (let key in clientConfig.options) {
+    const configKey = key
+      .replace('-', '')
+      .replace('_', '')
+      .toLowerCase();
+    const text = `${configKey}: ${clientConfig.options[key]}\n`;
+    configTxt = configTxt.concat(text);
+  }
+  if (!fs.existsSync(clientsPath)) {
+    logger.warn(
+      `Could not find requested client directory at ${clientsPath}. Creating new one...`
+    );
+    fs.mkdirpSync(clientsPath);
+  }
+  fs.writeFileSync(`${clientsPath}/${clientConfig.str('id')}.conf`, configTxt);
+  return clientConfig;
+}
+
+/*
+ * Retrieve a config from clients directory
+ * @param {string} id - id of client to retrieve
+ * @returns {bcfg.Config} - bcfg object of config
+ */
+
+function getConfig(id) {
+  assert(typeof id === 'string', 'Client config must have an id');
+
+  const appConfig = loadConfig('bpanel');
+  const clientsDir = appConfig.str('clients-dir', 'clients');
+
+  const clientsPath = resolve(appConfig.prefix, clientsDir);
+  const config = loadConfig(id, { id, prefix: clientsPath });
+
+  const path = resolve(clientsPath, `${id}.conf`);
+  if (!fs.existsSync(path)) {
+    const error = new Error(`File ${id}.conf not found`);
+    error.code = 'ENOENT';
+    throw error;
+  }
+
+  config.open(`${id}.conf`);
+  return config;
+}
+
+/*
+ * Simple utility that deletes a config
+ * Does not throw errors if client is not found
+ * NOTE: This does not confirm. Deletion is final!
+ * @param {string} id
+ * @returns {bool} - returns true when operation completed successfully
+ */
+
+function deleteConfig(id) {
+  assert(typeof id === 'string', 'Expected to get id of config to delete');
+  const config = getConfig(id);
+  const path = resolve(config.prefix, `${config.str('id')}.conf`);
+  const exists = fs.existsSync(path);
+  if (!exists)
+    logger.info('Attempt to delete config failed because file does not exist');
+  else {
+    try {
+      fs.unlinkSync(path);
```

Comment:
Wouldn't this fail if the file doesn't exist as well? The control flow would be more simple I think, definitely less nested without the `if/else->try/catch`

Reply:
One of the reasons for this was similar to above, basically just custom log messages for the server without throwing errors. I was also thinking about saving the deletion operation if it wasn't necessary assuming that's more costly. Although, I guess you could argue that it still might be better to do 1 deletion that fails than possibly 2 async actions.

So maybe:
```
// remove `exists`
let success = true;
try { 
  fs.unlinkSync(path);
} catch(e) {
  logger.error('Problem removing file:', e);
  success = false;
}
return success;
```

Reply:
updated to the above code snippet, let me know what you think.

Reply:
I think its possible to do this without the intermediate variable `success`

```javascript
try { 
  fs.unlinkSync(path);
  return true;
} catch(e) {
  logger.error('Problem removing file:', e);
  return false;
}
```